internal/api/authz/token.go

```diff
@@ -55,19 +71,21 @@ func Start(authZRepo authZRepo, issuer string, keys map[string]*SystemAPIUser) (
 			1*time.Hour,
 			time.Second,
 		),
-	}
+		systemUsers: systemUsers,
+	}, nil
 }
 
-func (v *TokenVerifier) VerifyAccessToken(ctx context.Context, token string, method string) (userID, clientID, agentID, prefLang, resourceOwner string, err error) {
-	if strings.HasPrefix(method, "/zitadel.system.v1.SystemService") {
-		userID, err := v.verifySystemToken(ctx, token)
-		if err != nil {
-			return "", "", "", "", "", err
-		}
-		return userID, "", "", "", "", nil
-	}
+func (v *TokenVerifier) VerifyAccessToken(ctx context.Context, token string) (userID, clientID, agentID, prefLang, resourceOwner string, isSystemUser bool, err error) {
 	userID, agentID, clientID, prefLang, resourceOwner, err = v.authZRepo.VerifyAccessToken(ctx, token, "", GetInstance(ctx).ProjectID())
-	return userID, clientID, agentID, prefLang, resourceOwner, err
+	if err == nil || !zitadel_errors.IsUnauthenticated(err) {
```

Comment:
Conditional which mean that either you have no error or you have another error than unauthenticated?
Is this condition meant to be like that?

Reply:
Yes. Only if verifying the access token returns an unauthenticated error we want to check if system token verification succeeds.